src/main/scala/kamon/jmx/collector/SupportedKamonMetricTypes.scala

```diff
@@ -1,57 +1,63 @@
 package kamon.jmx.collector
 
-import kamon.{Kamon, Tags}
-import kamon.metric.{Metric, CounterMetric, HistogramMetric, GaugeMetric}
+import kamon.Kamon
+import kamon.metric.Instrument
+import kamon.tag.TagSet
 
 private[collector] object SupportedKamonMetricTypes {
 
   trait SupportedKamonMetricType {
-    type T <: Metric[_]
+    type T <: Instrument[_, _]
     protected def getMetricInstrument(metricName: String): T
-    private[collector] def recordValue(metricInstrument: T, value: Long, tags: Tags = Map.empty): Unit
+    private[collector] def recordValue(metricInstrument: T, value: Long, tags: TagSet = TagSet.Empty): Unit
 
-    def record(metricName: String, value: Long, tags: Tags = Map.empty): Unit = recordValue(getMetricInstrument(metricName), value, tags)
+    def record(metricName: String, value: Long, tags: TagSet = TagSet.Empty): Unit = recordValue(getMetricInstrument(metricName), value, tags)
   }
 
   case object Counter extends SupportedKamonMetricType {
-    override type T = CounterMetric
-    override protected def getMetricInstrument(metricName: String): CounterMetric = Kamon.counter(metricName)
-    override private[collector] def recordValue(metricInstrument: CounterMetric, value: Long, tags: Tags = Map.empty): Unit =
-      metricInstrument.refine(tags).increment(value)
+    override type T = kamon.metric.Counter
+    override protected def getMetricInstrument(metricName: String): T = Kamon.counter(metricName).withoutTags()
+    override private[collector] def recordValue(metricInstrument: T, value: Long, tags: TagSet = TagSet.Empty): Unit =
+      metricInstrument.withTags(tags).increment(value)
   }
 
   case object Histogram extends SupportedKamonMetricType {
-    override type T = HistogramMetric
-    override protected def getMetricInstrument(metricName: String): HistogramMetric = Kamon.histogram(metricName)
-    override private[collector] def recordValue(metricInstrument: HistogramMetric, value: Long, tags: Tags = Map.empty): Unit =
-      metricInstrument.refine(tags).record(value)
+    override type T = kamon.metric.Histogram
```

Comment:
Not a biggy, just suggestion, maybe we can rename this into `import kamon.metric.{Histogram => KamonHistogram}` or something so we don't need to use the full package name; what do you think?

Reply:
but then I would have to do that for the other metrics (Counter, Gauge, RangeSampler)...

Reply:
Yeah I think it's preference: either use the full package name (that is explicit but can be long) or rename classes? I prefer to rename, but it's not a big deal.